lms/djangoapps/certificates/models.py

```diff
@@ -558,6 +563,9 @@ def certificate_status(generated_certificate):
         if generated_certificate.status == CertificateStatuses.downloadable:
             cert_status['download_url'] = generated_certificate.download_url
 
+        if generated_certificate.course_id.certificate_available_date < datetime.datetime.utcnow():
```

Comment:
I could move the date-checking into a method associated with the actual model, which upon checking the date also changes the status in the model. If the current date isn't after the availability date, then I'd return date_unavailable. But if the current date IS after the availability date, is there some sort of default "ok" state that we can change it back to? It looks like there's no one central place, like certificate_status_for_student(), where the GeneratedCertificate statuses are being changed, so I'm not sure about the logic for changing the state back after it is available...